packages/babel-parser/src/parser/statement.js

```diff
@@ -1498,13 +1498,16 @@ export default class StatementParser extends ExpressionParser {
   }
 
   pushClassProperty(classBody: N.ClassBody, prop: N.ClassProperty) {
-    // This only affects properties, not methods.
-    if (this.isNonstaticConstructor(prop)) {
+    if (
+      !prop.computed &
+      (prop.key.name === "constructor" || prop.key.value === "constructor")
```

Comment:
Could you add a comment that the second test is needed for strings?

Reply:
WDYT?

Reply:
Maybe like it's in the original function:
```js
  isNonstaticConstructor(method: N.ClassMethod | N.ClassProperty): boolean {
    return (
      !method.computed &&
      !method.static &&
      (method.key.name === "constructor" || // Identifier
        method.key.value === "constructor") // String literal
    );
  }
```